client/driver/docker.go

```diff
@@ -83,28 +88,37 @@ func NewDockerDriver(ctx *DriverContext) Driver {
 // to connect to the docker daemon. In production mode we will read
 // docker.endpoint from the config file.
 func (d *DockerDriver) dockerClient() (*docker.Client, error) {
-	// Default to using whatever is configured in docker.endpoint. If this is
-	// not specified we'll fall back on NewClientFromEnv which reads config from
-	// the DOCKER_* environment variables DOCKER_HOST, DOCKER_TLS_VERIFY, and
-	// DOCKER_CERT_PATH. This allows us to lock down the config in production
-	// but also accept the standard ENV configs for dev and test.
-	dockerEndpoint := d.config.Read("docker.endpoint")
-	if dockerEndpoint != "" {
-		cert := d.config.Read("docker.tls.cert")
-		key := d.config.Read("docker.tls.key")
-		ca := d.config.Read("docker.tls.ca")
-
-		if cert+key+ca != "" {
-			d.logger.Printf("[DEBUG] driver.docker: using TLS client connection to %s", dockerEndpoint)
-			return docker.NewTLSClient(dockerEndpoint, cert, key, ca)
-		} else {
-			d.logger.Printf("[DEBUG] driver.docker: using standard client connection to %s", dockerEndpoint)
-			return docker.NewClient(dockerEndpoint)
+	if client != nil {
+		return client, nil
+	}
+
+	var err error
+	createClient.Do(func() {
```

Comment:
> I thought we ran fingerprinters periodically, though. If so it could fail if nomad starts before Docker, and would then succeed on a subsequent fingerprint pass.

I'm currently using Systemd to make sure Nomad is started after Docker. The same could be done in Upstart. I think that's a fair assumption.